templates/project/build.rs

```diff
@@ -1,16 +1,59 @@
 use cargo_metadata::{CargoOpt, DependencyKind, MetadataCommand};
 use regex::Regex;
+use serde::{Deserialize, Serialize};
 use std::env;
 
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Config {
+    pub cloud: Cloud,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct Cloud {
+    pub id: String,
+    pub secret: String,
+    pub app_address: String,
+}
+
 fn main() {
     if Regex::new(r"\w+-esp3?2?s?\d?-espidf")
         .unwrap()
         .is_match(&env::var("TARGET").unwrap())
     {
-        embuild::build::CfgArgs::output_propagated("MICRO_RDK").unwrap();
-        embuild::build::LinkArgs::output_propagated("MICRO_RDK").unwrap();
+        println!("cargo:rerun-if-changed=viam.json");
+        println!("cargo:rerun-if-env-changed=MICRO_RDK_WIFI_SSID");
+        println!("cargo:rerun-if-env-changed=MICRO_RDK_WIFI_PASSWORD");
+        if env::var("TARGET").unwrap() == "xtensa-esp32-espidf" {
```

Comment:
I think this isn't quite right to be filtering on this target. Doesn't this mean that if you built, for, say esp32s3 that you wouldn't get values set for wifi ssid / password even if they were in the environment?

And, actually, now that I think about it: I don't see any real reason that handing the SSID  and PASSWORD needs to be gated on `TARGET` at all. That is really just here to gate doing the `embuild` stuff. In which case, I'd argue for the same change to micro-rdk-server/build.rs

Reply:
removed the conditional for TARGET and QEMU from project template build.rs